src/etos_test_runner/lib/workspace.py

```diff
@@ -160,6 +160,13 @@ def test_directory(
                 chdir(self.identifiers.get(identifier))
             with self.collect_logs(self.identifiers.get(identifier)):
                 yield self.identifiers.get(identifier)
+        except:  # pylint:disable=bare-except
+            # Should something raise an exception we shall assume that the workspace became dirty
+            # and we remove it from the stored workspaces so that it may not be used again.
+            # We avoid cleaning up the workspace so that it will be saved in the workspace archive
+            # for investigative purposes.
```

Comment:
The comment i skind of confusing, as to what it is we're actually removing.

On the one hand we say that we're removing dirty workspaces, and athe saem time we say that we avoid cleaning up the workspace so that it can be kept for further invesigations. It sound kind of contradictove to me.

Reply:
The difference is that in the first part we say that we just remove it from the 'stored' workspaces so that it may not be used again by another test and in the second case we say that we don't clean it up.
In code, we remove if from the list of stored workspaces but we don't clean it up.

I will clarify